### DISCUSSION TITLE
How to create camera trajectory path visualization?

### DISCUSSION THREAD

**Discussion by preethamam** [isl-org/Open3D#3739] [Q&A]
I have searched the Open3D tutorials but did not find a solution to the camera trajectory visualization similar to the image below: 
<img width="1223" alt="Screen Shot 2021-07-15 at 2 09 17 AM" src="https://user-images.githubusercontent.com/28588878/125762229-1bb769ff-86b7-490d-a9ea-0dfd859d92f8.png">

Any help is appreciated.

**Comment by edgarriba** [comment]
I’m looking also for this. Any update on this ?

**Comment by yxlao** [comment]
Hi, cameras can be visualized as LineSets. You'll need to compute the camera center and draw a "camera box" from it. Check out this unofficial example code: https://github.com/yxlao/Open3D/blob/realsense-record/examples/Python/ReconstructionSystem/visualize_pose.py 

We are planning to add native camera/trajectory plotting support in Open3D.

**Reply by edgarriba** [reply]
thanks so much! One more question, do you have anywhere an example to render in "real time" and update the geometries ? I'm prototyping a feature based slam example with [kornia](https://github.com/kornia/kornia) and wanted to demo the output of the camera and sparse 3d reconstruction over time.

**Reply by germanros1987** [reply]
@edgarriba you can also look at: [staticcreate_camera_visualization](http://www.open3d.org/docs/latest/python_api/open3d.geometry.LineSet.html#open3d.geometry.LineSet.create_camera_visualization)

**Reply by edgarriba** [reply]
thanks!
